src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,347 @@
+          }
+          e.style.bottom = `${
+            (tooltipHeight - offsetForTooltipBodyHeight) / 2 +
+            offsetForBottomMargin +
+            offsetForBottomPadding
+          }px`
+        }
+
+        /**
+         * We try to set the position based on the
+         * original intention, but make adjustments
+         * if the element is clipped out of the viewport
+         */
+        switch (position) {
+          case 'top':
+            positionTop(tooltipBody)
+            if (!isElementInViewport(tooltipBody)) {
+              positionBottom(tooltipBody)
+            }
+            break
+          case 'bottom':
+            positionBottom(tooltipBody)
+            if (!isElementInViewport(tooltipBody)) {
+              positionTop(tooltipBody)
+            }
+            break
+          case 'right':
+            positionRight(tooltipBody)
+            if (!isElementInViewport(tooltipBody)) {
+              positionLeft(tooltipBody)
+              if (!isElementInViewport(tooltipBody)) {
+                positionTop(tooltipBody)
+              }
+            }
+            break
+          case 'left':
+            positionLeft(tooltipBody)
+            if (!isElementInViewport(tooltipBody)) {
+              positionRight(tooltipBody)
+              if (!isElementInViewport(tooltipBody)) {
+                positionTop(tooltipBody)
+              }
+            }
+            break
+
+          default:
+            // skip default case
+            break
+        }
+      }
+    }, [isVisible])
+  }
+
+  if (isCustomProps(props)) {
+    const triggerElementRef = useRef<HTMLElement>(null)
+    const { label, position, asCustom, children, ...remainingProps } = props
+    const customProps: FCProps = (remainingProps as unknown) as FCProps
+
+    const tooltipClasses = classnames('usa-tooltip__body', {
+      'is-set': isVisible,
+      'usa-tooltip__body--top': position === 'top',
+      'usa-tooltip__body--bottom': position === 'bottom',
+      'usa-tooltip__body--right': position === 'right',
+      'usa-tooltip__body--left': position === 'left',
+      'is-visible': isVisible,
+    })
+    const activateTooltip = (): void => {
+      setVisible(true)
+    }
+    const deactivateTooltip = (): void => {
+      setVisible(false)
+    }
+
+    useTooltip(triggerElementRef, position)
+
+    const triggerElement = createElement(
+      asCustom,
+      {
+        ref: triggerElementRef,
+        'data-testid': 'triggerElement',
+        onMouseEnter: () => activateTooltip(),
+        onMouseOver: () => activateTooltip(),
+        onFocus: () => activateTooltip(),
+        onMouseLeave: () => deactivateTooltip(),
+        onBlur: () => deactivateTooltip(),
+        onKeyDown: () => deactivateTooltip(),
+        ...customProps,
+      },
+      children
+    )
+
+    return (
+      <span
+        data-testid="tooltipWrapper"
+        ref={wrapperRef}
+        className="usa-tooltip"
+        {...customProps}
```

Comment:
`customProps` are passed into the `triggerElement` below, so they should not be repeated on this wrapper span